server/enhancedDate.js

```diff
@@ -1 +1,127 @@
-// export this as a Node module using the enhancedDate.js code from Module_2
\ No newline at end of file
+/* 
+ * add your module definition for enhancedDate here
+ * assign the module to a global variable
+ * Date object: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date
+ *
+ *
+ * setDate: Stores the passed-in date. 
+ * Takes milliseconds after epoch or a date object. 
+ * Do some type-checking to make sure you have a valid type. 
+ * If no value is passed, default to now
+ *
+ * getDate: Returns either milliseconds after epoch by default 
+ * or a date object if true is passed as an argument.
+ *
+ * getDayName: Returns the full day name as a string, e.g. "Monday", "Tuesday", etc.
+ *
+ * getMonthName: Return the full month name as a string, e.g. "January", "February", etc.
+ *
+ * isFuture: Return boolean true if date is in the future (from when method is called)
+ *
+ * isToday: Return boolean true if the stored date is "today".
+ */
+
+//var enhancedDate = (function () {
+    // private data variables
+    var myDate;
+    
+    var setDate = function (dateItem) {
+        //console.log("setDate");
+        if (dateItem === undefined) {
+            myDate = new Date();
+        } else {
+            myDate = new Date(dateItem);
+        }
+    };
+    
+    var getDate = function (dateItem) {
```

Comment:
All other methods that aren't setDate should check if myDate is set before trying to use it and call setDate if it hasn't been.